lib/filterx/expr-condition.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2023 shifter <shifter@axoflow.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx/expr-condition.h"
+#include "filterx/object-primitive.h"
+
+struct _FilterXCondition
+{
+  FilterXExpr super;
+  FilterXExpr *condition_expr;
+  GList *if_branch;
+  FilterXCondition *else_branch;
+};
```

Comment:
Naming ideas:

We call the same thing `conditional`, not `condition` in the simple filter code, and I think it suits better.

Maybe we should have something like this:
```c
struct _FilterXConditional
{
  FilterXExpr super;
  FilterXExpr *condition;
  GList *statements;
  FilterXCondition *false_branch;
};
```

This `Conditional` expression has a `condition`. If it evaluates to true, the expression's `statements` get evaluated, otherwise we continue with the `false_branch`.